pool/app/custom_field/repo/repo_partial_update.ml

```diff
@@ -0,0 +1,227 @@
+open Entity
+module Database = Pool_database
+module Dynparam = Utils.Database.Dynparam
+
+let update_sihl_user pool ?firstname ?lastname contact =
+  let open CCOption in
+  let ctx = Pool_tenant.to_ctx pool in
+  let given_name = firstname <+> contact.Contact.user.Sihl_user.given_name in
+  let name = lastname <+> contact.Contact.user.Sihl_user.name in
+  Service.User.update ~ctx ?given_name ?name contact.Contact.user
+;;
+
+let update_sql column_fragment =
+  let base = {sql| UPDATE pool_contacts SET profile_updated_at = $2, |sql} in
+  let where_fragment =
+    {sql| WHERE user_uuid = UNHEX(REPLACE($1, '-', '')) |sql}
+  in
+  Format.asprintf "%s %s %s" base column_fragment where_fragment
+;;
+
+let upsert_answer_request =
+  let open Caqti_request.Infix in
+  {sql|
+    INSERT INTO pool_custom_field_answers (
+      uuid,
+      custom_field_uuid,
+      entity_uuid,
+      value,
+      version
+    ) VALUES (
+      UNHEX(REPLACE($1, '-', '')),
+      UNHEX(REPLACE($2, '-', '')),
+      UNHEX(REPLACE($3, '-', '')),
+      $4,
+      $5
+    )
+    ON DUPLICATE KEY UPDATE
+      value = VALUES(value),
+      version = VALUES(version)
+    |sql}
+  |> Repo_entity_answer.Write.t ->. Caqti_type.unit
+;;
+
+let upsert_admin_answer_request =
+  let open Caqti_request.Infix in
+  {sql|
+  INSERT INTO pool_custom_field_answers (
+    uuid,
+    custom_field_uuid,
+    entity_uuid,
+    admin_value,
+    admin_version
+  ) VALUES (
+    UNHEX(REPLACE($1, '-', '')),
+    UNHEX(REPLACE($2, '-', '')),
+    UNHEX(REPLACE($3, '-', '')),
+    $4,
+    $5
+  )
+  ON DUPLICATE KEY UPDATE
+    admin_value = VALUES(admin_value),
+    admin_version = VALUES(admin_version)
+  |sql}
+  |> Repo_entity_answer.Write.t ->. Caqti_type.unit
+;;
+
+let clear_answer_request is_admin =
+  let open Caqti_request.Infix in
+  let open Pool_common.Repo in
+  let where =
+    {sql|
+    WHERE
+      custom_field_uuid = UNHEX(REPLACE($1, '-', ''))
+        AND
+      entity_uuid = UNHEX(REPLACE($2, '-', ''))
+    |sql}
+  in
+  let update =
+    match is_admin with
+    | false ->
+      {sql|
+      UPDATE pool_custom_field_answers
+        SET
+          value = NULL,
+          version = version + 1
+    |sql}
+    | true ->
+      {sql|
+      UPDATE pool_custom_field_answers
+        SET
+          admin_value = NULL,
+          admin_version = admin_version + 1
+    |sql}
+  in
+  Format.asprintf "%s %s" update where
+  |> Caqti_type.(tup2 Id.t Id.t) ->. Caqti_type.unit
```

Comment:
There might be others, where this could be changed as well

```suggestion
  |> Caqti_type.(tup2 Id.t Id.t ->. unit)
```

Reply:
I bet 😄  found a couple